### PR TITLE
CI: Merge PyPi workflows

### LINKED ISSUE #90: Merge PyPi workflows
At the moment we have two almost identical workflows for publishing to the different PyPi servers. We do want to unify them to have less duplicated configuration files.

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes #90

### CODE REVIEW

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          python3-pip
+          python3-libvirt
+          python3-sphinx
+          python3-sphinx-rtd-theme
+      - name: Install dependencies
+        run: pip3 install .[lint,test,docs]
+      - name: Fetch date for version bump
+        run: echo "new_version=$(date +'%Y%m%d%H%M')" >> $GITHUB_ENV
+      - name: Replace version in setup.py
+        if: startsWith(github.ref, 'refs/tags/v') != true && github.ref == 'refs/heads/master'
```
Comment (tiltingpenguin) [inline .github/workflows/pypi.yml:31]: Shouldn't this be migrated to main?

File: .github/workflows/pypi.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        # Only replace the version if we are not on a tag and on master.
+        run: sed -i '/__version__ = "[0-9].[0-9].[0-9]/s/.$/.'${{ env.new_version }}'"/g' koan/__init__.py
+      - name: Build a binary wheel and a source tarball
+        run: make release
+      - name: Publish distribution to Test PyPI
+        if: github.ref == 'refs/heads/master'
```
Comment (tiltingpenguin) [inline .github/workflows/pypi.yml:37]: Another master reference that should be main

**Review by tiltingpenguin — CHANGES_REQUESTED** [review]

File: .github/workflows/pypi.yml (hunk 1 above)
Comment (SchoolGuy) [reply .github/workflows/pypi.yml]: Many thanks. That would have been overlooked.

File: .github/workflows/pypi.yml (hunk 2 above)
Comment (SchoolGuy) [reply .github/workflows/pypi.yml]: Many thanks. That would have been overlooked.

**Review by tiltingpenguin — APPROVED** [review]
LGTM

**Pull request merged**